src/mjolnir/ingest_transit.cc

```diff
@@ -312,7 +312,7 @@ write_stops(Transit& tile, const tile_transit_info_t& tile_info, feed_cache_t& f
     int node_count = tile.nodes_size();
     for (const auto& child : tile_children) {
       auto child_stop = feed.get_stop(child.second);
-      if (child.first.id == station.id &&
+      if (child.first.id == station.id && child.first.feed == station.feed &&
```

Comment:
was a small bug: when there's 2 feeds which have the same stop id's in the both feeds, it previously would duplicate nodes for each feed.